tests/unit_tests/data_steward/utils/bq_test.py

```diff
@@ -0,0 +1,88 @@
+from utils.bq import define_dataset, update_labels_and_tags
+
+
+class BqTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        # Input parameters expected by the class
+        self.project_id = 'bar_project'
+        self.dataset_id = 'foo_dataset'
+        self.description = 'fake_description'
+        self.existing_labels_or_tags = {'label': 'value', 'tag': ''}
+        self.new_labels_or_tags = {'label': 'new_value', 'new_tag': ''}
+        self.updated = {'tag': '', 'label': 'new_value', 'new_tag': ''}
+
+    def test_define_dataset(self):
+        # Tests if project_id is given
+        self.assertRaises(RuntimeError, define_dataset, None, self.dataset_id,
+                          self.description, self.existing_labels_or_tags)
+
+        # Tests if dataset_id is given
+        self.assertRaises(RuntimeError, define_dataset, self.project_id, None,
+                          self.description, self.existing_labels_or_tags)
+
+        # Tests if description is given
+        self.assertRaises(RuntimeError, define_dataset, self.project_id,
+                          self.dataset_id, (None or ''),
+                          self.existing_labels_or_tags)
+
+        # Tests if no label or tag is given
+        self.assertRaises(RuntimeError, define_dataset, self.project_id,
+                          self.dataset_id, self.description, None)
+
+        # Pre-conditions
+        results = define_dataset(self.project_id, self.dataset_id,
+                                 self.description, self.existing_labels_or_tags)
+
+        # Post conditions
+        self.assertIsInstance(results, bigquery.Dataset)
+        self.assertEqual(results.labels, self.existing_labels_or_tags)
+
+    def test_update_labels_and_tags(self):
+        # Tests if dataset_id param is provided
+        self.assertRaises(RuntimeError, update_labels_and_tags, None,
+                          self.existing_labels_or_tags, self.new_labels_or_tags)
+
+        # Tests if new_labels_or_tags param is provided
+        self.assertRaises(RuntimeError, update_labels_and_tags, self.dataset_id,
+                          self.existing_labels_or_tags, None)
+
+        # Pre-conditions
+        results_1 = update_labels_and_tags(self.dataset_id,
+                                           self.existing_labels_or_tags,
+                                           self.new_labels_or_tags, True)
+
+        results_2 = update_labels_and_tags(self.dataset_id,
+                                           self.existing_labels_or_tags,
+                                           self.existing_labels_or_tags, False)
+
+        # Post conditions
+        self.assertEqual(results_1, self.updated)
+        self.assertRaises(RuntimeError, results_2)
```

Comment:
I think we should test that an error is thrown when a label is overwritten and `overwrite_ok==False`.
```suggestion
        with self.assertRaises(RuntimeError):
            update_labels_and_tags(self.dataset_id,
                                   existing_labels_or_tags={'label': 'apples'},
                                   new_labels_or_tags={'label': 'oranges'},
                                   overwrite_ok=False)
```